### PR TITLE
Feat/add support for batch write

### PR DESCRIPTION
**Pull request by jens-peterolsson**
This code is mostly copied from other usage and refactored. The retry code is still a bit messy IMO, but I'll take opinions on whether we should update it or not.

No support for deletes, will come in a later PR.

### CODE REVIEW

**Review by Tanax — APPROVED** [review]

File: src/randomizer/random-integer.test.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+describe('random assignment', () => {
+  it('should assign randomly and evenly', () => {
+    const summary = {
+      lessThan500: 0,
+      atOrOver500: 0
+    };
+
+    for (let ix = 0; ix < 1000; ix += 1) {
+      const r = randomInteger(1, 1000);
```
Comment (Crafoord) [inline src/randomizer/random-integer.test.js:11]: How can we be certain that this test always work? I mean, it involves random and we send in the same props 1000 times.

We should add a test that it always returns a value between min and max.

Comment (jens-peterolsson) [reply src/randomizer/random-integer.test.js:11]: Agree about adding a more specific test, will do that.

The thinking about this test is that if the spread is wider than 60/40 over 1000 iterations then the algorithm is not good enough, it should be close to 50/50. You're right though that this mechanism opens up for edge cases where the test in theory might pass 999 times out of 1000 for instance. It would also be clearer if the test only used 2 distinct values, but that IMO is too far from the expected usage. I'd still say 60/40 is conservative enough and 1000 iterations is more than enough for it to always pass and that testing the expected usage is worth the imperfect setup (not possible to have an exact setup since it's random of course and mocking random would defeat the purpose). Don't agree?

Comment (Crafoord) [reply src/randomizer/random-integer.test.js:11]: I guess that makes sense to test that the algorithm does what you expect. Just not a fan of a test possibly failing a deploy but as you say, really unlikely if it performs as expected.

**Review by Crafoord — APPROVED** [review]
